x-pack/plugins/fleet/server/services/epm/packages/install_state_machine/_state_machine_package_install.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type {
+  ElasticsearchClient,
+  Logger,
+  SavedObject,
+  SavedObjectsClientContract,
+  ISavedObjectsImporter,
+} from '@kbn/core/server';
+import { SavedObjectsErrorHelpers } from '@kbn/core/server';
+
+import type { IAssignmentService, ITagsClient } from '@kbn/saved-objects-tagging-plugin/server';
+
+import { PackageSavedObjectConflictError } from '../../../../errors';
+
+import type { HTTPAuthorizationHeader } from '../../../../../common/http_authorization_header';
+import { INSTALL_STATES } from '../../../../../common/types';
+import type { PackageInstallContext, StateNames, StateContext } from '../../../../../common/types';
+import type { PackageAssetReference } from '../../../../types';
+
+import type {
+  Installation,
+  InstallType,
+  InstallSource,
+  PackageVerificationResult,
+  EsAssetReference,
+  KibanaAssetReference,
+  IndexTemplateEntry,
+  AssetReference,
+} from '../../../../types';
+
+import {
+  stepCreateRestartInstallation,
+  stepInstallKibanaAssets,
+  stepInstallILMPolicies,
+  stepInstallMlModel,
+  stepInstallIndexTemplatePipelines,
+  stepRemoveLegacyTemplates,
+  stepUpdateCurrentWriteIndices,
+  stepInstallTransforms,
+  stepDeletePreviousPipelines,
+  stepSaveArchiveEntries,
+  stepSaveSystemObject,
+  updateLatestExecutedState,
+} from './steps';
+import type { StateMachineDefinition } from './state_machine';
+import { handleState } from './state_machine';
+
+export interface InstallContext extends StateContext<StateNames> {
+  savedObjectsClient: SavedObjectsClientContract;
+  savedObjectsImporter: Pick<ISavedObjectsImporter, 'import' | 'resolveImportErrors'>;
+  savedObjectTagAssignmentService: IAssignmentService;
+  savedObjectTagClient: ITagsClient;
+  esClient: ElasticsearchClient;
+  logger: Logger;
+  installedPkg?: SavedObject<Installation>;
+  packageInstallContext: PackageInstallContext;
+  installType: InstallType;
+  installSource: InstallSource;
+  spaceId: string;
+  force?: boolean;
+  verificationResult?: PackageVerificationResult;
+  authorizationHeader?: HTTPAuthorizationHeader | null;
+  ignoreMappingUpdateErrors?: boolean;
+  skipDataStreamRollover?: boolean;
+
+  indexTemplates?: IndexTemplateEntry[];
+  packageAssetRefs?: PackageAssetReference[];
+  // output values
+  esReferences?: EsAssetReference[];
+  kibanaAssetPromise?: Promise<KibanaAssetReference[]>;
+}
+
+export async function _stateMachineInstallPackage(
```

Comment:
I think this function could use a documentation style comment that explains each object property and how the state machine works as a whole.